src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -342,7 +342,17 @@ public virtual void AddSubSystem(String name, ISubSystem subsystem)
 			}
 			else if (name == SubSystemConstants.NamingKey)
 			{
-				NamingSubSystem = (INamingSubSystem)subsystem;
+				var namingSubSystem = subsystem as INamingSubSystem;
+
+				if (NamingSubSystem != null)
+				{
+					foreach (var handler in NamingSubSystem.GetAllHandlers())
+					{
+						namingSubSystem?.Register(handler);
+					}
```

Comment:
The problem with this is that a naming subsystem is more than just handlers, this doesn't account for handler selectors and filters. Even though the MicroKernel (part of Windsor now) exposes `INamingSubSystem` and `AddSubSystem` is definitely hasn't been an extension point that has been used much or at all so it is possible we'll run into other areas where things don't work quite right.

Could we achieve the ordering of handlers somewhere else rather than a custom naming subsystem?